site/en/tutorials/load_data/images.ipynb

```diff
@@ -0,0 +1,1356 @@
+        "\n",
+        "The simple pipeline used above reads each file individually, on each epoch. This is fine for local training on CPU but may not be sufficient for GPU training, and is totally inapprpriate for any sort of distributed training. "
+      ]
+    },
+    {
+      "metadata": {
+        "id": "oNmQqgGhLWie",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "To investigate, first build a simple function to check the performance of our datasets:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "_gFVe1rp_MYr",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "import time\n",
+        "\n",
+        "def timeit(ds, batches=100):\n",
+        "  overall_start = time.time()\n",
+        "  # Fetch a single batch to prime the pipeline (fill the shuffle buffer),\n",
+        "  # before starting the timer\n",
+        "  it = iter(ds.take(batches+1))\n",
+        "  next(it)\n",
+        "\n",
+        "  start = time.time()\n",
+        "  for i,(images,labels) in enumerate(it):\n",
+        "    if i%10 == 0:\n",
+        "      print('.',end='')\n",
+        "  print()\n",
+        "  end = time.time()\n",
+        "\n",
+        "  duration = end-start\n",
+        "  print(\"{} batches: {} s\".format(batches, duration))\n",
+        "  print(\"{:0.5f} Images/s\".format(BATCH_SIZE*batches/duration))\n",
+        "  print(\"Total time: {}s\".format(end-overall_start))"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "TYiOr4vdLcNX",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "The performance of the current dataset is:"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "qwOJGxw_FkJU",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "ds = image_label_ds.repeat()\n",
+        "ds = ds.shuffle(buffer_size=4000) \n",
+        "ds = ds.batch(BATCH_SIZE).prefetch(1)\n",
+        "ds"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "Su_P8s8ZA41P",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "timeit(ds)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "HsLlXMO7EWBR",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "### Cache"
+      ]
+    },
+    {
+      "metadata": {
+        "id": "lV1NOn2zE2lR",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Use `tf.data.Dataset.cache` to easily take advantage of the performance boost of working with in memory data.\n",
```

Comment:
Make note of the fact that `cache` is only beneficial if the input dataset fits into memory. If it does not, then using `cache` does more harm then good.

Reply:
nevermind, you already talk about this later